lib/install/validate-tree.js

```diff
@@ -73,5 +74,31 @@ function thenCheckTop (idealTree, next) {
     warnObj.code = 'ENODEPRE'
     idealTree.warnings.push(warnObj)
   }
+
+  next()
+}
+
+// check if dependencies are duplicated between dependencies and
+// dev-dependencies. Issue warning if so.
+function thenCheckDuplicateDeps (idealTree, next) {
+  var shorterDepsList = idealTree.package.devDependencies
+  var longerDepsList = idealTree.package.dependencies
+  // might be undefined -- skip if so
+  if (!(idealTree.package.devDependencies === undefined || idealTree.package.dependencies === undefined)) {
```

Comment:
I would prefer to just set defaults for the deplists and skip this check, eg:

`var shorterDepsList = idealTree.package.devDependencies || {}`